Translation of deploying-to-netlify.md (#163)
gatsbyjs/gatsby-pt-BR: docs/docs/deploying-to-netlify.md

---
title: Publicando no Netlify
---

> Esse guia explica como publicar e hospedar seu próximo _site_ do Gatsby no [Netlify](https://www.netlify.com/).

Reviewer: Adotamos a convenção de "_site_ Gatsby", então, acho que é melhor mantermos aqui também:

```suggestion
Esse guia explica como publicar e hospedar seu próximo _site_ Gatsby no [Netlify](https://www.netlify.com/).
```

> Netlify é uma opção excelente para publicar _sites_ Gatsby. Netlify é uma plataforma unificada que automatiza seu código para criar _sites_ e aplicativos _web_ de fácil manutenção e alta performance. Eles fornecem publicação contínua (Acionadas por construção no Git); um inteligente CDN global; DNS completo (incluindo domínios customizados); HTTPS automatizado; Aceleração de _asset_; e muito mais.

Reviewer 2: ```suggestion
Netlify é uma opção excelente para publicar _sites_ Gatsby. Netlify é uma plataforma unificada que automatiza seu código para criar _sites_ e aplicativos _web_ de fácil manutenção e alta performance. Eles fornecem publicação contínua (Acionadas por construção no Git); um inteligente CDN global; DNS completo (incluindo domínios customizados); HTTPS automatizado; Aceleração de _assets_; e muito mais.
```

> Seu nível gratuito inclui projetos pessoais e comerciais ilimitados, HTTPS, publicação contínua de repositórios públicos ou privados e mais.

Reviewer 2: ```suggestion
Eles fornecem um nível gratuito que inclui projetos pessoais e comerciais ilimitados, HTTPS, publicação contínua de repositórios públicos ou privados e mais.
```
Talvez em pt fique melhor assim :). O que achas?

## Configuração de hospedagem

Existem dois meios de hospedar o seu _site_:

1. [Configuração de repositório do Git](#configuração-de-repositório-do-git)

2. [Enviando a pasta do _site_](#enviando-a-pasta-do-site)

### Configuração de repositório do Git

> Netlify tem suporte embutido para [Github](https://github.com/), [Gitlab](https://about.gitlab.com/) e [Bitbucket](https://bitbucket.org/). Essa abordagem permite reverter para versões antigas do _site_ sempre que você desejar. Você ainda ganha a habilidade de publicar novamente o _site_ enviando o código ao repositório respectivo, sem a necessidade de fazer a construção e o envio de forma manual toda vez que você realizar alterações. Seu repositório pode ser privado ou público.

Reviewer: ```suggestion
Netlify tem suporte embutido para [Github](https://github.com/), [GitLab](https://about.gitlab.com/) e [Bitbucket](https://bitbucket.org/). Essa abordagem permite reverter para versões antigas do _site_ sempre que você desejar. Você ainda ganha a habilidade de publicar novamente o _site_ enviando o código ao repositório respectivo, sem a necessidade de fazer a construção e o envio de forma manual toda vez que você realizar alterações. Seu repositório pode ser privado ou público.
```

> Netlify tem suporte embutido para [Github](https://github.com/), [Gitlab](https://about.gitlab.com/) e [Bitbucket](https://bitbucket.org/). Essa abordagem permite reverter para versões antigas do _site_ sempre que você desejar. Você ainda ganha a habilidade de publicar novamente o _site_ enviando o código ao repositório respectivo, sem a necessidade de fazer a construção e o envio de forma manual toda vez que você realizar alterações. Seu repositório pode ser privado ou público.

Reviewer 2: ```suggestion
Netlify tem suporte embutido para [Github](https://github.com/), [Gitlab](https://about.gitlab.com/) e [Bitbucket](https://bitbucket.org/). Essa abordagem permite reverter para versões antigas do _site_ sempre que você desejar. Você ainda ganha a habilidade de publicar novamente o _site_ enviando o código ao respectivo repositório, sem a necessidade de fazer a construção e o envio de forma manual toda vez que você realizar alterações, sendo seu repositório público ou privado.
```
Acho que fica mais fluido em pt

> Agora, entre no netlify e você vai ver um botão `Novo site do git` no canto superior direito da sua tela. Clique nele e conecte com o mesmo provedor git que você usa para hospedar o seu _website_ e permita o Netlify a usar sua conta. Escolha o repositório do seu _website_ e você será levado as configurações de publicação com as opções abaixo.

Reviewer: Acho que faz sentido manter o título original do botão, já que o site não tem versão em Portugues, e explicar o que o botão faz. Por exemplo:

```suggestion
Agora, entre no netlify e você vai ver um botão `New site from git` (Novo site a partir do git) no canto superior direito da sua tela. Clique nele e conecte com o mesmo provedor git que você usa para hospedar o seu _website_ e permita ao Netlify usar sua conta. Escolha o repositório do seu _website_ e você será levado as configurações de publicação com as opções abaixo.
```

Também sugeri uma pequena alteração mais abaixo.

> Agora, entre no netlify e você vai ver um botão `Novo site do git` no canto superior direito da sua tela. Clique nele e conecte com o mesmo provedor git que você usa para hospedar o seu _website_ e permita o Netlify a usar sua conta. Escolha o repositório do seu _website_ e você será levado as configurações de publicação com as opções abaixo.

Reviewer 2: ```suggestion
Agora, entre no netlify e você vai ver um botão `Novo site do git` no canto superior direito da sua tela. Clique nele e conecte com o mesmo provedor git que você usa para hospedar o seu _site_ e permita o Netlify a usar sua conta. Escolha o repositório do seu _site_ e você será levado as configurações de publicação com as opções abaixo.
```
Acredito que seja melhor manter um padrão no documento, então aconselho o uso ou de "website" ou de "site" no texto inteiro

> - Branch para publicação: Você pode especificar uma _branch_ para ser monitorada. Quando você fizer o envio para aquela _branch_ específica, somente então o Netlify irá construir e publicar o _site_. A _branch_ padrão é a `master`.

Reviewer: Mesma coisa acima, acho que vale a pena manter em Inglês e Português

```suggestion
- _Branch to Deploy_ (Branch para publicação): Você pode especificar uma _branch_ para ser monitorada. Quando você fizer o envio para aquela _branch_ específica, somente então o Netlify irá construir e publicar o _site_. A _branch_ padrão é a `master`.
```

> - Comando para construção: Você pode especificar o comando que você quer que o Netlify rode quando você enviar para a _branch_ acima. O comando padrão é `npm run build`.

Reviewer: ```suggestion
- _Build Command_ (Comando para construção): Você pode especificar o comando que você quer que o Netlify rode quando você enviar para a _branch_ acima. O comando padrão é `npm run build`.
```

> - Diretório de publicação: Você pode especificar qual pasta o Netlify deve usar para hospedar seu _website_, exemplo, _public_, _dist_, _build_. O diretório padrão é o `public`.

Reviewer: ```suggestion
- _Publish directory_ (Diretório de publicação): Você pode especificar qual pasta o Netlify deve usar para hospedar seu _website_, exemplo, _public_, _dist_, _build_. O diretório padrão é o `public`.
```

> - Configurações avançadas de construção: Se o _site_ necessita de variáveis de ambiente para construção, você especificá-las clicando em `Mostrar avançado` e então no botão `Nova variável`. 

Reviewer: ```suggestion
- Configurações avançadas de construção: Se o _site_ necessita de variáveis de ambiente para construção, você especificá-las clicando em _Show advanced_ (Mostrar avançado) e então no botão _New variable_ (Nova variável). 
```

> Clique no botão `Publicar site` e o Netlify vai começar o processo de construção e publicação que foi especificado por você. Você pode ir na aba `Publicados` e ver o processo se desenrolar no `Log de publicação`. Após alguns momentos, você vai receber o URL em que seu _site_ foi publicado, exemplo `nome-aleatorio.netlify.com`

Reviewer: ```suggestion
Clique no botão _Deploy site_ (Publicar site) e o Netlify vai começar o processo de construção e publicação que foi especificado por você. Você pode ir na aba _Deploys_ (Publicados) e ver o processo se desenrolar no _Deploy log_ (Log de publicação). Após alguns momentos, você vai receber o URL em que seu _site_ foi publicado, exemplo `nome-aleatorio.netlify.com`
```

### Enviando a pasta do site

Existe também a opção de enviar o seu _site_ para o Netlify sem utilizar git.

> Para a [_build_ de produção](/docs/glossary#build), você vai precisar rodar o comando `gatsby build`; Gatsby vai gerar o _site_ de produção na pasta `public`. Durante o processo de construção o CSS, JavaScript, HTML e as imagens vão ser otimizados e colocados dentro dessa pasta.

Reviewer 2: ```suggestion
Para a [_build_ de produção](/docs/glossary#build), você vai precisar rodar o comando `gatsby build`; Gatsby vai gerar o _site_ de produção na pasta `public`. Durante o processo de construção, o CSS, JavaScript, HTML e as imagens vão ser otimizados e colocados dentro dessa pasta.
```

```shell
gatsby build
```

> Quando a construção estiver completa, você está pronto para enviar seu site ao Netlify. Vá em [Netlify](https://app.netlify.com/) e faça o _login_ ou se cadastre usando qualquer método. Após realizar o _login_ com sucesso, você verá a mensagem mostrada abaixo:

Reviewer 2: ```suggestion
Quando a construção estiver completa, você estará pronto para enviar seu site ao Netlify. Vá em [Netlify](https://app.netlify.com/) e faça o _login_ ou se cadastre usando qualquer método. Após realizar o _login_ com sucesso, você verá a mensagem mostrada abaixo:
```

> ```text

Reviewer: Mantem a mensagem em inglês, e coloca a embaixo em parenteses, que nem acima. O que acha? Aqui talvez possa ficar como você fez

Author: Eu acho que pode ficar as duas, tanto em inglês quanto traduzida pro português. Deixava a tradução em parênteses embaixo mesmo da original

    Você quer publicar um novo site sem conectar ao Git?
          Arraste e solte a pasta do seu site aqui
```

> Para iniciar o processo de publicação, você precisa apenas arrastar e soltar a pasta `public` sobre a área acima no _site_ do Netlify. Netlify vai criar um novo _site_ com um nome aleatório, e então começar a enviar e hosperdar os arquivos da aplicação. Após alguns momentos, ele vai lhe disponibilizar um URL para o _site_, exemplo `nome-aleatorio.netlify.com`

Reviewer 2: ```suggestion
Para iniciar o processo de publicação, você precisa apenas arrastar e soltar a pasta `public` sobre a área acima no _site_ do Netlify. Netlify vai criar um novo _site_ com um nome aleatório, e então começar a enviar e hospedar os arquivos da aplicação. Após alguns momentos, ele vai lhe disponibilizar um URL para o _site_, exemplo `nome-aleatorio.netlify.com`
```

![alt text](./images/gatsby-default-starter.png "Starter padrão do Gatsby")

## Publicação contínua

> Agora que o seu _site_ está conectado ao seu repositório, Netlify vai fazer a publicação sempre que você enviar ao seu repositório Git.

Reviewer 2: ```suggestion
Agora que o seu _site_ está conectado ao seu repositório, o Netlify vai fazer a publicação sempre que você enviar alterações para o seu repositório Git.
```

## Configuração de domínio

> Na `Visão geral` do seu _site_, você pode ir em `Configurações de domínio`. Ao adicionar um domínio personalizado e definir o registro `CNAME` como o URL do projeto Netlify nas configurações do seu provedor de DNS, você deve ser capaz de ver o projeto do Netlify na URL de seu domínio.

Reviewer: ```suggestion
Na aba _Overview_ (Visão geral) do seu _site_, você pode ir em _Domain Settings_ (Configurações de domínio). Ao adicionar um domínio personalizado e definir o registro `CNAME` como o URL do projeto Netlify nas configurações do seu provedor de DNS, você deve ser capaz de ver o projeto do Netlify na URL de seu domínio.
```

---
title: Publicando no Netlify
---

Esse guia explica como publicar e hospedar seu próximo _site_ Gatsby no [Netlify](https://www.netlify.com/).

> Netlify é uma opção excelente para publicar _sites_ Gatsby. Netlify é uma plataforma unificada que automatiza seu código para criar _sites_ e aplicativos _web_ de fácil manutenção e alta performance. Eles fornecem publicação contínua (Acionadas por construção no Git); um inteligente CDN global; DNS completo (incluindo domínios customizados); HTTPS automatizado; Aceleração de _assets_; e muito mais.

Reviewer 3: ```suggestion
Netlify é uma opção excelente para publicar _sites_ Gatsby. Netlify é uma plataforma unificada que automatiza seu código para criar _sites_ e aplicativos _web_ de fácil manutenção e alta performance. Eles fornecem publicação contínua (a partir de _builds_ acionados através do Git); um  CDN global inteligente; DNS completo (incluindo domínios customizados); HTTPS automatizado; Aceleração de _assets_; e muito mais.
```

Reviewer 3: Voltando agora depois de ler tudo vi que foi "build" traduzido para "construção" no artigo inteiro. Talvez pelo menos na primeira vez que é citado valha a pena incluir em parênteses o termo original em ingles? 
Incidentalmente: Particularmente não gosto de build traduzido como "construção" no contexto desse artigo.

Author: Build é um termo que eu sempre fico na dúvida se deve ou não ser traduzido. Por uma preferência pessoal, eu acredito que seja melhor deixar _build_ em inglês mesmo, não é um termo complicado de se entender, porém a tradução se torna estranha considerando as opções disponíveis para se traduzir _build_.

Reviewer 3: Confesso que não sei como ficou traduzido nos outros artigos, talvez alguém mais possa opinar. Para não "segurar" a tradução sugiro vc só mencionar mesmo o termo em inglês na primeira vez que citar.

Author: Ok, irei realizar as demais alterações e envio mais tarde. Obrigado

## Configuração de hospedagem

Existem dois meios de hospedar o seu _site_:

1. [Configuração de repositório do Git](#configuração-de-repositório-do-git)

2. [Enviando a pasta do _site_](#enviando-a-pasta-do-site)

### Configuração de repositório do Git

Netlify tem suporte embutido para [Github](https://github.com/), [GitLab](https://about.gitlab.com/) e [Bitbucket](https://bitbucket.org/). Essa abordagem permite reverter para versões antigas do _site_ sempre que você desejar. Você ainda ganha a habilidade de publicar novamente o _site_ enviando o código ao respectivo repositório, sem a necessidade de fazer a construção e o envio de forma manual toda vez que você realizar alterações, sendo seu repositório público ou privado.

Agora, entre no netlify e você vai ver um botão `New site from git` (Novo site a partir do git) no canto superior direito da sua tela. Clique nele e conecte com o mesmo provedor git que você usa para hospedar o seu _site_ e permita ao Netlify usar sua conta. Escolha o repositório do seu _site_ e você será levado as configurações de publicação com as opções abaixo.

> - _Branch to Deploy_ (Branch para publicação): Você pode especificar uma _branch_ para ser monitorada. Quando você fizer o envio para aquela _branch_ específica, somente então o Netlify irá construir e publicar o _site_. A _branch_ padrão é a `master`.

Reviewer 3: ```suggestion
- _Branch to Deploy_ (Branch para publicação): Você pode especificar uma _branch_ para ser monitorada. Quando você fizer o envio para aquela _branch_ específica, só então o Netlify irá construir e publicar o _site_. A _branch_ padrão é a `master`.
```

- _Build Command_ (Comando para construção): Você pode especificar o comando que você quer que o Netlify rode quando você enviar para a _branch_ acima. O comando padrão é `npm run build`.
- _Publish directory_ (Diretório de publicação): Você pode especificar qual pasta o Netlify deve usar para hospedar seu _site_, exemplo, _public_, _dist_, _build_. O diretório padrão é o `public`.
- Configurações avançadas de construção: Se o _site_ necessita de variáveis de ambiente para construção, você especificá-las clicando em _Show advanced_ (Mostrar avançado) e então no botão _New variable_ (Nova variável). 

> Clique no botão _Deploy site_ (Publicar site) e o Netlify vai começar o processo de construção e publicação que foi especificado por você. Você pode ir na aba _Deploys_ (Publicados) e ver o processo se desenrolar no _Deploy log_ (Log de publicação). Após alguns momentos, você vai receber o URL em que seu _site_ foi publicado, exemplo `nome-aleatorio.netlify.com`

Reviewer 3: ```suggestion
Clique no botão _Deploy site_ (Publicar site) e o Netlify vai começar o processo de construção e publicação que foi especificado por você. Você pode ir na aba _Deploys_ (Publicados) e acompanhar o progresso do processo no _Deploy log_ (Log de publicação). Após alguns momentos, você vai receber o URL em que seu _site_ foi publicado, exemplo `nome-aleatorio.netlify.com`
```

### Enviando a pasta do site

Existe também a opção de enviar o seu _site_ para o Netlify sem utilizar git.

> Para a [_build_ de produção](/docs/glossary#build), você vai precisar rodar o comando `gatsby build`; Gatsby vai gerar o _site_ de produção na pasta `public`. Durante o processo de construção, o CSS, JavaScript, HTML e as imagens vão ser otimizados e colocados dentro dessa pasta.

Reviewer 3: ```suggestion
Para a [_build_ de produção](/docs/glossary#build), você vai precisar rodar o comando `gatsby build`; Gatsby vai gerar o _site_ de produção na pasta `public`. Durante o processo de construção, o CSS, JavaScript, HTML e as imagens vão ser otimizados e incluídos nessa pasta.
```

```shell
gatsby build
```

> Quando a construção estiver completa, você estará pronto para enviar seu site ao Netlify. Vá em [Netlify](https://app.netlify.com/) e faça o _login_ ou se cadastre usando qualquer método. Após realizar o _login_ com sucesso, você verá a mensagem mostrada abaixo:

Reviewer 3: ```suggestion
Quando a construção estiver completa, você estará pronto para enviar seu site ao Netlify. Acesse [Netlify](https://app.netlify.com/) e faça o _login_ ou se cadastre usando qualquer método. Após realizar o _login_ com sucesso, você verá a mensagem abaixo:
```

```text
    Want to deploy a new site without connecting to Git?
          Drag and drop your site folder here
    (Você quer publicar um novo site sem conectar ao Git?
          Arraste e solte a pasta do seu site aqui)
```

> Para iniciar o processo de publicação, você precisa apenas arrastar e soltar a pasta `public` sobre a área acima no _site_ do Netlify. Netlify vai criar um novo _site_ com um nome aleatório, e então começar a enviar e hospedar os arquivos da aplicação. Após alguns momentos, ele vai lhe disponibilizar um URL para o _site_, exemplo `nome-aleatorio.netlify.com`

Reviewer 3: ```suggestion
Para iniciar o processo de publicação, você precisa apenas arrastar e soltar a pasta `public` sobre a área acima no _site_ do Netlify. Netlify vai criar um novo _site_ com um nome aleatório, e então começar a enviar e hospedar os arquivos da aplicação. Após alguns momentos será gerado um URL para o _site_, exemplo `nome-aleatorio.netlify.com`
```

![alt text](./images/gatsby-default-starter.png "Starter padrão do Gatsby")

## Publicação contínua

Agora que o seu _site_ está conectado ao seu repositório, o Netlify vai fazer a publicação sempre que você enviar alterações para o seu repositório Git.

## Configuração de domínio

> Na aba _Overview_ (Visão geral) do seu _site_, você pode ir em _Domain Settings_ (Configurações de domínio). Ao adicionar um domínio personalizado e definir o registro `CNAME` como o URL do projeto Netlify nas configurações do seu provedor de DNS, você deve ser capaz de ver o projeto do Netlify na URL de seu domínio.

Reviewer 3: ```suggestion
Na aba _Overview_ (Visão geral) do seu _site_, você pode acessar a opção _Domain Settings_ (Configurações de domínio). Ao adicionar um domínio personalizado e definir o registro `CNAME` como o URL do projeto Netlify nas configurações do seu provedor de DNS, você deve ser capaz de ver o projeto do Netlify na URL de seu domínio.
```
